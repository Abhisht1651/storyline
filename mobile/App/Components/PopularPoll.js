// @flow
import React from 'react'
import { View, Text, Image } from 'react-native'

// Styles
import styles from './Styles/PopularPollStyle'
import images from '../Themes/Images'

type Props = {
  border: boolean,
  question: string,
  votes: number,
  timeLeft: number,
  isPollActive: boolean,
  voterIcons: Array<string>,
  progressBar: Array<{ name: string, percentage: string }>,
  activeProgressBar: number
}

const popularPoll = (props: Props) => {
  const {
    border,
    question,
    votes,
    timeLeft,
    isPollActive,
    voterIcons,
    progressBar,
    activeProgressBar
  } = props
  const { pollActiveIcon, pollInactiveIcon } = images
  const pollIcon = isPollActive ? pollActiveIcon : pollInactiveIcon
  const pollText = isPollActive ? `${timeLeft} hours left` : 'Poll ended'
  const fillColor = isPollActive ? '#7B46E4' : '#47C3F4'

  const progressBarStyles = [
    { ...styles.progressFill, width: progressBar[0].percentage },
    { ...styles.progressFill, width: progressBar[1].percentage },
    { ...styles.progressFill, width: progressBar[2].percentage },
    { ...styles.progressFill, width: progressBar[3].percentage }
  ]

  progressBarStyles[activeProgressBar] = {
    ...progressBarStyles[activeProgressBar],
    backgroundColor: fillColor
  }

  const progressBarActivePercentageText = { ...styles.progressBarPercentageText, color: '#FFFFFF' }
  const progressBarActiveText = { ...styles.progressBarText, color: '#FFFFFF' }

  const voterIconsElement = voterIcons.map((voterIcon, index) => (
    <Image source={voterIcon} key={index} style={styles.voterIcon} />
  ))

  return (
    <View style={border ? styles.container : styles.containerWithoutBorder}>
      <View style={styles.questionContainer}>
        <Text style={styles.questionText}>{question}</Text>
      </View>
      <View style={styles.voterAndVotesContainer}>
        <View style={styles.voterContainer}>
          <View style={styles.voterIconContainer}>{voterIconsElement}</View>
          <View style={styles.votesContainer}>
            <Text style={styles.votes}>{votes}K votes</Text>
          </View>
        </View>
        <View style={styles.timeContainer}>
          <View style={styles.timeIconContainer}>
            <Image source={pollIcon} style={styles.pollIcon} />
          </View>
          <View style={styles.timeTextContainer}>
            <Text style={styles.timeText}>{pollText}</Text>
          </View>
        </View>
      </View>

      <View style={styles.progressBarContainer}>
        <View style={styles.progressBar}>
          <View style={progressBarStyles[0]} />
          <View style={styles.progressBarTextContainer}>
            <Text
              style={
                activeProgressBar === 0
                  ? progressBarActivePercentageText
                  : styles.progressBarPercentageText
              }>
              {progressBar[0].percentage}
            </Text>
            <Text style={activeProgressBar === 0 ? progressBarActiveText : styles.progressBarText}>
              {progressBar[0].name}
            </Text>
          </View>
          {activeProgressBar === 0 ? (
            <View style={styles.progressBarVoterIconContainer}>
              <Image source={voterIcons[0]} style={styles.progressBarVoterIcon} />
            </View>
          ) : null}
        </View>
        <View style={styles.progressBar}>
          <View style={progressBarStyles[1]} />
          <View style={styles.progressBarTextContainer}>
            <Text
              style={
                activeProgressBar === 1
                  ? progressBarActivePercentageText
                  : styles.progressBarPercentageText
              }>
              {progressBar[1].percentage}
            </Text>
            <Text style={activeProgressBar === 1 ? progressBarActiveText : styles.progressBarText}>
              {progressBar[1].name}
            </Text>
            {activeProgressBar === 1 ? (
              <View style={styles.progressBarVoterIconContainer}>
                <Image source={voterIcons[0]} style={styles.progressBarVoterIcon} />
              </View>
            ) : null}
          </View>
        </View>
        <View style={styles.progressBar}>
          <View style={progressBarStyles[2]} />
          <View style={styles.progressBarTextContainer}>
            <Text
              style={
                activeProgressBar === 2
                  ? progressBarActivePercentageText
                  : styles.progressBarPercentageText
              }>
              {progressBar[2].percentage}
            </Text>
            <Text style={activeProgressBar === 2 ? progressBarActiveText : styles.progressBarText}>
              {progressBar[2].name}
            </Text>
            {activeProgressBar === 2 ? (
              <View style={styles.progressBarVoterIconContainer}>
                <Image source={voterIcons[0]} style={styles.progressBarVoterIcon} />
              </View>
            ) : null}
          </View>
        </View>
        <View style={styles.progressBar}>
          <View style={progressBarStyles[3]} />
          <View style={styles.progressBarTextContainer}>
            <Text
              style={
                activeProgressBar === 3
                  ? progressBarActivePercentageText
                  : styles.progressBarPercentageText
              }>
              {progressBar[3].percentage}
            </Text>
            <Text style={activeProgressBar === 3 ? progressBarActiveText : styles.progressBarText}>
              {progressBar[3].name}
            </Text>
            {activeProgressBar === 3 ? (
              <View style={styles.progressBarVoterIconContainer}>
                <Image source={voterIcons[0]} style={styles.progressBarVoterIcon} />
              </View>
            ) : null}
          </View>
        </View>
      </View>
    </View>
  )
}

popularPoll.defaultProps = {
  border: true,
  question: 'Who are your favorites for the World Cup 2020?',
  votes: 54,
  timeLeft: 2,
  isPollActive: true,
  voterIcons: [images.voterIcon1, images.voterIcon2, images.voterIcon3],
  progressBar: [
    { name: 'India', percentage: '53%' },
    { name: 'Australia', percentage: '12%' },
    { name: 'West Indies', percentage: '6%' },
    { name: 'England', percentage: '29%' }
  ],
  activeProgressBar: 0
}

export default popularPoll
