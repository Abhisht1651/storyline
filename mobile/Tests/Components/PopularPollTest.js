// @flow
import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import PopularPoll from '../../App/Components/PopularPoll'

describe('PopularPoll', () => {
  it('PopularPoll component renders correctly', () => {
    const wrapper = shallow(<PopularPoll />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<PopularPoll />)
    expect(wrapper.find('Text')).toHaveLength(11)
  })
  it('should render the Image Component', () => {
    const wrapper = shallow(<PopularPoll />)
    expect(wrapper.find('Image')).toHaveLength(5)
  })
})
