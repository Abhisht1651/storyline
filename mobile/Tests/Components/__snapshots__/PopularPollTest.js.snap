// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularPoll PopularPoll component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#E6ECED",
      "borderRadius": 30,
      "borderStyle": "solid",
      "borderWidth": 1,
      "minHeight": 345,
      "minWidth": 300,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 26,
        "width": "88%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Medium",
          "fontSize": 36,
          "letterSpacing": -0.11,
          "lineHeight": 44,
          "opacity": 1,
          "textAlign": "left",
        }
      }
    >
      Who are your favorites for the World Cup 2020?
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#E6ECED",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 24,
        "paddingBottom": 10,
        "width": "88%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginLeft": 8,
          }
        }
      >
        <Image
          key="0"
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "borderRadius": 28,
              "height": 56,
              "marginLeft": -16,
              "opacity": 1,
              "width": 56,
            }
          }
        />
        <Image
          key="1"
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "borderRadius": 28,
              "height": 56,
              "marginLeft": -16,
              "opacity": 1,
              "width": 56,
            }
          }
        />
        <Image
          key="2"
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "borderRadius": 28,
              "height": 56,
              "marginLeft": -16,
              "opacity": 1,
              "width": 56,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#082366",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 28,
              "letterSpacing": -0.1,
              "opacity": 0.5,
              "textAlign": "left",
            }
          }
        >
          54
          K votes
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 34,
              "width": 34,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 6,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#002468",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 24,
              "letterSpacing": -0.07,
              "opacity": 0.5,
              "textAlign": "right",
            }
          }
        >
          2 hours left
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 15,
        "marginTop": 20,
        "width": "86.67%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#E6ECED",
          "borderRadius": 4,
          "height": 56,
          "marginBottom": 15,
          "minWidth": "100%",
          "opacity": 1,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#7B46E4",
            "borderRadius": 4,
            "height": 56,
            "left": -4,
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "width": "53%",
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 24,
              "letterSpacing": -0.7,
              "opacity": 1,
              "paddingLeft": 10,
              "paddingVertical": 12,
              "textAlign": "left",
            }
          }
        >
          53%
        </Text>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 26,
              "letterSpacing": -0.7,
              "opacity": 1,
              "paddingLeft": 6,
              "paddingVertical": 12,
              "textAlign": "left",
            }
          }
        >
          India
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 24,
            "height": 48,
            "justifyContent": "center",
            "marginTop": 3.5,
            "opacity": 1,
            "position": "absolute",
            "right": 8,
            "width": 48,
          }
        }
      >
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "height": 40,
              "width": 40,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#E6ECED",
          "borderRadius": 4,
          "height": 56,
          "marginBottom": 15,
          "minWidth": "100%",
          "opacity": 1,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#47C3F4",
            "borderRadius": 4,
            "height": 56,
            "left": -4,
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "width": "12%",
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 24,
              "letterSpacing": -0.7,
              "opacity": 1,
              "paddingLeft": 10,
              "paddingVertical": 12,
              "textAlign": "left",
            }
          }
        >
          12%
        </Text>
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 26,
              "letterSpacing": -0.7,
              "opacity": 1,
              "paddingLeft": 6,
              "paddingVertical": 12,
              "textAlign": "left",
            }
          }
        >
          Australia
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#E6ECED",
          "borderRadius": 4,
          "height": 56,
          "marginBottom": 15,
          "minWidth": "100%",
          "opacity": 1,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#47C3F4",
            "borderRadius": 4,
            "height": 56,
            "left": -4,
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "width": "6%",
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 24,
              "letterSpacing": -0.7,
              "opacity": 1,
              "paddingLeft": 10,
              "paddingVertical": 12,
              "textAlign": "left",
            }
          }
        >
          6%
        </Text>
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 26,
              "letterSpacing": -0.7,
              "opacity": 1,
              "paddingLeft": 6,
              "paddingVertical": 12,
              "textAlign": "left",
            }
          }
        >
          West Indies
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#E6ECED",
          "borderRadius": 4,
          "height": 56,
          "marginBottom": 15,
          "minWidth": "100%",
          "opacity": 1,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#47C3F4",
            "borderRadius": 4,
            "height": 56,
            "left": -4,
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "width": "29%",
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 24,
              "letterSpacing": -0.7,
              "opacity": 1,
              "paddingLeft": 10,
              "paddingVertical": 12,
              "textAlign": "left",
            }
          }
        >
          29%
        </Text>
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Bold",
              "fontSize": 26,
              "letterSpacing": -0.7,
              "opacity": 1,
              "paddingLeft": 6,
              "paddingVertical": 12,
              "textAlign": "left",
            }
          }
        >
          England
        </Text>
      </View>
    </View>
  </View>
</View>
`;
